### PR TITLE
store: remove TrieUpdate::range

### PR DESCRIPTION
**Pull request by mina86**
The TrieUpdate::range function is never used and makes the code
noticeably more complicated.  Remove it.

### CODE REVIEW

File: core/store/src/trie/update.rs [hunk 1]
```diff
@@ -200,32 +191,17 @@ impl<'a> Iterator for MergeIter<'a> {
...
-                p.extend_from_slice(end);
-                Some(p)
-            }
-            None => None,
-        };
-        trie_iter.seek_prefix(&start_offset)?;
-        let committed_iter = state_update.committed.range(start_offset.clone()..).map(
+        trie_iter.seek_prefix(prefix)?;
+        let prefix = prefix.to_vec();
+        let committed_iter = state_update.committed.range(prefix.clone()..).map(
```
Comment (matklad) [inline core/store/src/trie/update.rs:204]: Not entirely sure what's going on here, but, after just throwing random stuff at the wall, the following seems to compile, while avoding a copy:

```rust
.range::<[u8], _>((Bound::Included(prefix.as_slice()), Bound::Unbounded))
```

File: core/store/src/trie/update.rs [hunk 2]
```diff
@@ -241,29 +217,24 @@ impl<'a> TrieUpdateIterator<'a> {
...
     }
 }
 
 impl<'a> Iterator for TrieUpdateIterator<'a> {
     type Item = Result<Vec<u8>, StorageError>;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let stop_cond = |key: &[u8], prefix: &[u8], end_offset: &Option<Vec<u8>>| {
+        let stop_cond = |key: &[u8], prefix: &[u8]| {
             // TODO(mina86): Figure out if starts_with check is still necessary.
```
Comment (matklad) [inline core/store/src/trie/update.rs:236]: Unrelated to the PR, but it seems like the logic here can be simplified by removing `Ordering` and instead first unconditionally fetching the smaller key, and then doing the prefix check at the end as a guard.

**Review by matklad — APPROVED** [review]
LGTM!

This is a bit fiddly, so please check that, if you introduce some random off-by-one bug here, we have a test which kills the mutant.

File: core/store/src/trie/update.rs (hunk 2 above)
Comment (mina86) [reply core/store/src/trie/update.rs:240]: There’s in fact even better way to deal with this by creating a range iterator with upper bound set to smallest key larger then prefix which does not start with a prefix.  We’re already doing something similar in RocksDB.  See next_prefix function.  I have a couple WIP commits which improve this code some more.

**Pull request merged**